tests/capdpa_test.py

```diff
@@ -10,11 +10,13 @@ def check(self, a, b):
         return self.assertEqual(a, b,
             "Invalid entity:\n " + " ".join(difflib.unified_diff(str(a).splitlines(1), str(b).splitlines(1), "Expected", "Result")))
 
-    def load(self, spec):
-        return open("tests/data/" + spec, "r").read()
+    def load(self, spec, directory=""):
+        return open("tests/data/" + directory + "/" + spec, "r").read()
 
-    def cache(self, res, spec):
-        if not os.path.exists("tests/cache/"):
-            os.mkdir("tests/cache/")
-        return open("tests/cache/" + spec, "w").write(res)
+    def cache(self, res, spec, directory=""):
+        if not os.path.exists("tests/cache"):
+            os.mkdir("tests/cache")
+        if not os.path.exists("tests/cache/" + directory):
+            os.mkdir("tests/cache/" + directory)
```

Comment:
The check for `tests/cache` could be removed if we use `os.makedirs` instead as it works recursively.

Reply:
That's much nicer, indeed. Changed.